src/Form/components/ExtractionToast.tsx

```diff
@@ -0,0 +1,285 @@
+import React, { useState } from 'react';
+
+type DataItem = {
+  status: 'complete' | 'polling' | 'error' | 'queued';
+  extraction_key: string;
+  extraction_variant_key?: string;
+  children?: DataItem[];
+};
+
+type ExtractionToastProps = {
+  data: DataItem[];
+  title?: string;
+};
+
+const ExtractionItem = ({
+  item,
+  level = 0
+}: {
+  item: DataItem;
+  level?: number;
+}) => {
+  const [isExpanded, setIsExpanded] = useState(true);
+  const hasChildren = item.children && item.children.length > 0;
+
+  const paddingLeft = level * 24;
+
+  return (
+    <div css={{ width: '100%' }}>
+      <div
+        css={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'space-between',
+          padding: '8px 16px',
+          paddingLeft: `${paddingLeft + 16}px`,
+          cursor: hasChildren ? 'pointer' : 'default',
+          ':hover': {
+            backgroundColor: '#f9fafb'
+          }
+        }}
+        onClick={() => hasChildren && setIsExpanded(!isExpanded)}
+      >
+        <div
+          css={{
+            display: 'flex',
+            alignItems: 'center',
+            gap: '12px',
+            flex: 1
+          }}
+        >
+          <StatusIcon status={item.status} />
+          <span
+            css={{
+              color: item.status === 'queued' ? '#9ca3af' : '#374151',
+              fontWeight: 500,
+              fontSize: '14px'
+            }}
+          >
+            {item.extraction_key && item.extraction_variant_key
+              ? `${item.extraction_key} (${item.extraction_variant_key})`
+              : item.extraction_key ||
+                {
+                  complete: 'Completed Extraction',
+                  error: 'Failed Extraction',
+                  queued: 'Queued Extraction',
+                  polling: 'Running Extraction'
+                }[item.status] ||
+                'Extraction'}
+          </span>
+        </div>
+
+        <div
+          css={{
+            display: 'flex',
+            alignItems: 'center',
+            gap: '8px'
+          }}
+        >
+          {hasChildren && (isExpanded ? <ChevronUp /> : <ChevronDown />)}
+        </div>
+      </div>
+
+      {hasChildren && isExpanded && (
+        <div css={{ width: '100%' }}>
+          {item.children?.map((child, index) => (
+            <ExtractionItem key={index} item={child} level={level + 1} />
+          ))}
+        </div>
+      )}
+    </div>
+  );
+};
+
+const ExtractionToast = ({
```

Comment:
AIExtractionToast maybe? also should each component be pulled into its own file?